src/esp/assets/ResourceManager.cpp

```diff
@@ -1648,43 +1751,257 @@ void ResourceManager::loadMaterials(Importer& importer,
+
+    /////////////////
+    // set Phong attributes appropriately from precalculated value
+    ////////////////
+    // normal mapping is already present in copied array if present in
+    // original material.
+
+    // make sure not to re-add values that already exist in array, or
+    // new color creation will assert.
+    safeAppend(origMaterialData, newAttributes, MnMatAttr::Shininess,
+               shininess);
+    safeAppend(origMaterialData, newAttributes, MnMatAttr::AmbientColor,
+               ambientColor);
+    safeAppend(origMaterialData, newAttributes, MnMatAttr::DiffuseColor,
+               diffuseColor);
+    safeAppend(origMaterialData, newAttributes, MnMatAttr::SpecularColor,
+               specColor);
+
+    // texture transforms, if there's none the returned matrix is an
+    // identity only copy if we don't already have TextureMatrix
+    // attribute (if original pbrMaterial does not have that specific
+    // matrix)
+    ESP_WARNING() << "before PBR texture matrix addition";
+
+    if (!setTexMatrix &&
+        (!pbrMaterial.hasAttribute(MnMatAttr::TextureMatrix))) {
+      arrayAppend(newAttributes, {MnMatAttr::TextureMatrix,
+                                  pbrMaterial.commonTextureMatrix()});
+      setTexMatrix = true;
+    }
+
+    ESP_WARNING() << "after PBR texture matrix addition";
+
+    if (pbrMaterial.hasAttribute(MnMatAttr::BaseColorTexture)) {
+      // only provide texture indices if BaseColorTexture attribute
+      // exists
+      const Mn::UnsignedInt BCTexture = pbrMaterial.baseColorTexture();
+      safeAppend(origMaterialData, newAttributes, MnMatAttr::AmbientTexture,
+                 BCTexture);
+      safeAppend(origMaterialData, newAttributes, MnMatAttr::DiffuseTexture,
+                 BCTexture);
+      if (metalness >= 0.5) {
+        safeAppend(origMaterialData, newAttributes, MnMatAttr::SpecularTexture,
+                   BCTexture);
+      }
+    }
+  }  // if no phong material support exists in material
+
+  if (~(origMatTypes & Mn::Trade::MaterialType::PbrMetallicRoughness)) {
+    // add appropriate values for expected attributes for PbrMetallicRoughness
+    // derived from Phong attributes
+    const auto& phongMaterial =
+        origMaterialData->as<Mn::Trade::PhongMaterialData>();
+
+    // derive base color from Phong diffuse or ambient color, depending on which
+    // is present.  set to white if neither is present
+    const Mn::Color4 baseColor = phongMaterial.diffuseColor();
+
+    // Experimental metalness heuristic using saturation of spec color
+    // to derive approximation of metalness
+    const Mn::Color4 specColor = phongMaterial.specularColor();
+
+    // if specColor alpha == 0 then no metalness
+    float metalness = 0.0f;
+    // otherwise, this heuristic will derive a value for metalness based on
+    // how non-grayscale the specular color is (HSV Saturation).
+    if (specColor.a() != 0.0f) {
+      metalness = specColor.toHsv().saturation;
```

Comment:
```suggestion
      metalness = specColor.saturation();
```

avoids calculating the (to be thrown away) H and V values :)